test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -677,19 +684,34 @@ fi
 
 if [ "$patch_version" = true ]; then
     for b in ${patch_version_upgrade_downgrade_release_branches[@]}; do
-        # Build the tip of the release branch and run test to genearte data files
+        # Build the tip of the release branch and run test to generate data files
         (build_branch $b)
         (run_test_checkpoint "$b" "row")
 
         # Pick a patch version from the list of patch versions for the release branch
         cd $b; pv=$(pick_a_version $b); cd ..
+
         (build_branch $pv)
+        rtn=$(is_test_checkpoint_recovery_supported $pv)
+        patch_fix_included=$(git log --oneline --grep=WT-8708 -b "$pv" --)
+
+        # Apply patch fix from WT-8708 to already released compatible versions to avoid test/checkpoint setting commit timestamp less than stable timestamp
+        if [ $rtn == "yes" ] && [ ! $patch_fix_included ]; then
```

Comment:
In line 715 there's another checking of the `$rtn` value. Should we move this if logic into an elif before line 717, to make the logic more readable?

Reply:
I don't think we can as if in the case that `$rtn = "no"`, the elif won't get entered and the patch fix won't get applied.

Reply:
I might have not described it clearly. Here is the suggested change:
```
        if [ $rtn == "no" ]; then
            echo -e "\n\"$pv\" does not support test checkpoint with recovery, skipping ...\n"
        elif [ $rtn == "yes" ] && [ ! $patch_fix_included ]; then 
            ...
        else
            (verify_test_checkpoint "$pv" "$b" "row")
```

Reply:
Ah yes, thanks for clarifying, I misunderstood. Have changed it according to your suggestion!